riscv/triggers.cc

```diff
@@ -314,10 +314,8 @@ std::optional<match_result_t> itrigger_t::detect_trap_match(processor_t * const
 {
   state_t * const state = proc->get_state();
   if ((state->prv == PRV_M && !m) ||
-      (!state->v && state->prv == PRV_S && !s) ||
-      (!state->v && state->prv == PRV_U && !u) ||
-      (state->v && state->prv == PRV_S && !vs) ||
-      (state->v && state->prv == PRV_U && !vu)) {
+      (state->prv == PRV_S && !(state->v ? vs : s)) ||
+      (state->prv == PRV_U && !(state->v ? vu : u))) {
     return std::nullopt;
   }
```

Comment:
Please DRY up this duplicate code before changing it in two places at once. I'm embarassed that I let this copypasta get past my review of #1128. `itrigger_t::detect_trap_match()` and `etrigger_t::detect_trap_match()` differ by only one `if()` expression.

It might be best to move m/s/u/vs/vu all up to `trigger_t`, and for those trigger types that don't use those bits, their constructors would set them to appropriate fixed values.

Reply:
Add a trigger_t::mode_match() function for privilege mode checking.